pyfr/integrators/dual/pseudo/base.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+
+from pyfr.integrators.basecommon import BaseCommon
+from pyfr.util import proxylist
+
+
+class BaseDualPseudoIntegrator(BaseCommon, object):
+    formulation = 'dual'
+    aux_nregs = 0
+
+    def __init__(self, backend, systemcls, rallocs, mesh,
+                 initsoln, cfg, stepper_coeffs):
+        # Sanity checks
+        if self._controller_needs_lerrest and not self._stepper_has_lerrest:
+            raise TypeError('Incompatible stepper/controller combination')
+
+        self.backend = backend
+        self.rallocs = rallocs
+        self.isrestart = initsoln is not None
+        self.cfg = cfg
+
+        sect = 'solver-time-integrator'
+
+        self._dtaumin = 1.0e-12
+        self._dtau = self.cfg.getfloat(sect, 'pseudo-dt')
+
+        self.maxniters = self.cfg.getint(sect, 'pseudo-niters-max', 0)
+        self.minniters = self.cfg.getint(sect, 'pseudo-niters-min', 0)
+
+        if self.maxniters < self.minniters:
+            raise ValueError('The maximum number of pseudo-iterations must '
+                             'be greater than or equal to the minimum')
+
+        self._pseudo_residtol= self.cfg.getfloat(sect, 'pseudo-resid-tol')
+        self._pseudo_norm = self.cfg.get(sect, 'pseudo-resid-norm', 'l2')
+
+        if self._pseudo_norm not in {'l2', 'uniform'}:
+            raise ValueError('Invalid pseudo-residual norm')
+
+        # Amount of temp storage required by physical stepper
+        self._stepper_nregs = len(stepper_coeffs) - 1
+
+        # Determine the amount of temp storage required in total
+        self.nreg = (self._pseudo_stepper_nregs + self._stepper_nregs +
```

Comment:
Lets be consistent when it comes to if we suffix nreg with `s`.  (I think this issue predates this pull request and is probably my fault.)